### PR TITLE
8319973: AArch64: Save and restore FPCR in the call stub

### PR DESCRIPTION
**Pull request by theRealAph**
On AArch64 we don't save and restore the default floating-point control state when we enter and leave Java code. We really should, because if we're called via the JNI invocation interface with a weird FP control state we'll not be Java compatible.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8319973](https://bugs.openjdk.org/browse/JDK-8319973): AArch64: Save and restore FPCR in the call stub (**Bug** - P3)

### Reviewers
 * [Andrew Dinn](https://openjdk.org/census#adinn) (@adinn - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16637/head:pull/16637` \
`$ git checkout pull/16637`

Update a local copy of the PR: \
`$ git checkout pull/16637` \
`$ git pull https://git.openjdk.org/jdk.git pull/16637/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16637`

View PR using the GUI difftool: \
`$ git pr show -t 16637`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16637.diff">https://git.openjdk.org/jdk/pull/16637.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16637#issuecomment-1809881748)

### CODE REVIEW

**Review by adinn — APPROVED** [review]
All looks good.

File: src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp [hunk 1]
```diff
@@ -141,7 +141,8 @@ class StubGenerator: public StubCodeGenerator {
   //     [ return_from_Java     ] <--- sp
   //     [ argument word n      ]
   //      ...
-  // -27 [ argument word 1      ]
+  // -29 [ argument word 1      ]
+  // -28 [ saved Floating-point Control Register ]
```
Comment (tstuefe) [inline src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp:145]: Why the 1-word gap? Alignment?

**Review by tstuefe — APPROVED** [review]
Looks good, one question inline.

File: src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp (hunk 1 above)
Comment (theRealAph) [reply src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp:145]: Yes, exactly. AArch64 stack is 16-aligned, and this is enforced by hardware.

**Pull request closed without merging**